### PR TITLE
Use embeddable ocdav go micro service

### PR DESCRIPTION
**Pull request by butonic**
We now use the reva `pgk/micro/ocdav` package that implements a go micro compatible version of the ocdav service.

Requires https://github.com/cs3org/reva/pull/2665 and uses a replace until it is merged.

next steps I do not see as part of this PR:
- [ ] extract ocdav into a dedicated extension
- [ ] make all debug endpoints use micro http service and discover them via the micro registry

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiSpaces-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/10178/21/1) failed. Further test are cancelled...

File: proxy/pkg/proxy/proxy.go [hunk 1]
```diff
@@ -86,6 +88,7 @@ func NewMultiHostReverseProxy(opts ...Option) *MultiHostReverseProxy {
 	for _, pol := range options.Config.Policies {
 		for _, route := range pol.Routes {
 			rp.logger.Debug().Str("fwd: ", route.Endpoint)
+			// TODO check either Backend or Service is set
```
Comment (C0rby) [inline proxy/pkg/proxy/proxy.go:91]: Could you implement the TODO now? It doesn't sound too big or difficult.

Comment (kobergj) [reply proxy/pkg/proxy/proxy.go]: Fair enough :+1:  Done.

**Review by C0rby — APPROVED** [review]
LGTM

**Pull request merged**